tests/infra/consortium.py

```diff
@@ -324,14 +325,43 @@ def get_proposal(self, remote_node, proposal_id):
             assert r.status_code == http.HTTPStatus.OK.value
             return r.body.json()
 
-    def retire_node(self, remote_node, node_to_retire, timeout=10):
+    def retire_node(self, remote_node, node_to_retire, timeout=10, network=None):
+        pending = False
+        with remote_node.client(connection_timeout=timeout) as c:
+            r = c.get(f"/node/network/nodes/{node_to_retire.node_id}")
+            pending = r.body.json()["status"] == infra.node.State.PENDING.value
         LOG.info(f"Retiring node {node_to_retire.local_node_id}")
         proposal_body, careful_vote = self.make_proposal(
             "remove_node",
             node_id=node_to_retire.node_id,
         )
         proposal = self.get_any_active_member().propose(remote_node, proposal_body)
         self.vote_using_majority(remote_node, proposal, careful_vote)
+        if remote_node == node_to_retire:
```

Comment:
Could the following be done under `infra.network.retire_node()` instead to separate concern between the execution of the proposal (members) and the cleanup of retired nodes (operators)?

Reply:
Yes, probably, I will take a look.

Reply:
It largely works, the only thing is the need to return/pass whether the node was pending before we retired it, but that seems fine.